packages/website/ts/components/wallet/wallet.tsx

```diff
@@ -491,16 +508,56 @@ export class Wallet extends React.Component<WalletProps, WalletState> {
         if (_.isEmpty(tokenAddresses)) {
             return {};
         }
+        // for each input token address, search for the corresponding symbol in this.props.tokenByAddress, if it exists
+        // create a mapping from existing symbols -> address
+        const tokenAddressBySymbol = _.fromPairs(
+            _.compact(
+                _.map(tokenAddresses, address => {
+                    const tokenIfExists = _.get(this.props.tokenByAddress, address);
+                    if (!_.isUndefined(tokenIfExists)) {
+                        const symbol = tokenIfExists.symbol;
+                        // the crypto compare api doesn't understand 'WETH' so we need to replace it with 'ETH'
+                        const key = symbol === ETHER_TOKEN_SYMBOL ? ETHER_SYMBOL : symbol;
+                        return [key, address];
+                    } else {
+                        return undefined;
+                    }
+                }),
+            ),
+        );
```

Comment:
I think it would be simpler to write this as:

```
const tokenAddressBySymbol: {[symbol: string]: string} = {};
_.each(tokenAddresses, address => {
    const tokenIfExists = _.get(this.props.tokenByAddress, address);
    if (!_.isUndefined(tokenIfExists)) {
        const symbol = tokenIfExists.symbol;
        // the crypto compare api doesn't understand 'WETH' so we need to replace it with 'ETH'
       const key = symbol === ETHER_TOKEN_SYMBOL ? ETHER_SYMBOL : symbol;
       tokenAddressBySymbol[key] = address;
    }
});
```

As much as possible, let's try and avoid using more non-standard Lodash methods (`fromPairs`, `compact`, etc...) as they can reduce code readability.